src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceCreationService.kt

```diff
@@ -153,6 +154,18 @@ class LicenceCreationService(
 
     val nomisRecord = prisonerSearchApiClient.searchPrisonersByNomisIds(listOf(prisonNumber)).first()
 
+    if (nomisRecord.homeDetentionCurfewActualDate == null) {
+      error("HDC licence for ${nomisRecord.prisonerNumber} could not be created as it is missing a HDCAD")
+    }
+
+    if (nomisRecord.homeDetentionCurfewEligibilityDate == null) {
+      error("HDC licence for ${nomisRecord.prisonerNumber} could not be created as it is missing a HDCED")
+    }
+
+    if (!hdcService.isApprovedForHdc(nomisRecord.bookingId!!.toLong(), nomisRecord.homeDetentionCurfewEligibilityDate)) {
```

Comment:
cool, this doesn't do that address check though... 
might need to do that separately